task02/queue.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include "queue.h"
+#include <cstdlib>
+
+
+List* ListQueue::new_l()
+{
+       return new List;
+}
+
+void ListQueue::del_l(List *l)
+{
+	delete l->Data;
+	if (l->next != nullptr)
+	{
+		del_l(l->next);
+		delete l->next;
+	}
+}
+
+ListQueue::ListQueue()
+{
+	this->list = new_l();
+}
+
+ListQueue::~ListQueue()
+{
+	del_l(list);
+}
+
+void ListQueue::enqueue(data &d)
+{
+	if (list->Data == nullptr)
+	{
+		list->Data = new data;
```

Comment:
Do you really need `list->Data` as a pointer?
I strongly suggest declare to it as `data Data;`